test/lib.bash

```diff
@@ -170,6 +180,51 @@ function end_prober_test {
   wait ${PROBER_PID}
 }
 
+function run_knative_serving_operator_tests {
+  (
+  local version target serverless_rootdir exitstatus patchfile fork gitdesc
+  version="$1"
+  fork="${2:-knative}"
+  serverless_rootdir="$(pwd)"
+  make_temporary_gopath
+
+  logger.info "Checkout the code ${fork}/serving-operator @ ${version}"
+  mkdir -p "${GOPATH}/src/knative.dev"
+  target="${GOPATH}/src/knative.dev/serving-operator"
+  git clone --branch "${version}" --depth 1 \
+    "https://github.com/${fork}/serving-operator.git" \
+    "${target}"
+  pushd "${target}" || return $?
+
+  patchfile="${serverless_rootdir}/test/patches/SRVKS-241-knative-serving-operator-skip-configure.patch"
+  logger.info "Apply walkaround for SRVKS-241"
+  logger.debug "Patchfile is: ${patchfile}"
+  [ -f "${patchfile}" ] || return $?
+  patch --strip=0 < "${patchfile}" || return $?
+
+  gitdesc=$(git describe --always --tags --dirty)
+
+  exitstatus=0
+
+  logger.info "Run tests of knative/serving-operator @ ${gitdesc}"
+  env TEST_NAMESPACE='knative-serving' \
+    go test -v -tags=e2e -count=1 -timeout=30m -parallel=1 ./test/e2e \
+      --kubeconfig "$KUBECONFIG" \
+    || exitstatus=5$? && true
+
+  if (( !exitstatus )); then
+    logger.success 'Tests have passed'
+  else
+    logger.error 'Tests have failures!'
+  fi
+
+  logger.info "Removing GOPATH: ${GOPATH}"
+  rm -rf "${GOPATH}"
+  popd || return $?
```

Comment:
We're running in a subshell so this is probably not necessary?

Reply:
Yes. Not necessary, but a good practice to restore.